support/grunt/config.ts

```diff
@@ -7,253 +7,323 @@ export interface WebServerConfig {
+		site: [ 'site/**/*.ts', '!site/node_modules/**' ]
 	},
 
-	'widget-core': {
-		options: {
-			repo: 'dojo/widget-core'
+	tutorials: {
+		'dojo2-tutorials': {
+			src: 'site/source/tutorials',
+			dest: '<%= distDirectory %>/tutorials/assets'
 		}
 	},
 
-	widgets: {
-		options: {
-			repo: 'dojo/widgets'
+	/**
+	 * Host a local development server
+	 */
+	webserv: {
+		server: {
+			middleware
 		}
-	}
-};
-
-export const clean = {
-	api: [ '<%= tempDirectory %>' ],
-	dist: [ '<%= distDirectory %>' ],
-	publish: [ '<%= publishDirectory %>' ],
-	sync: [ '<%= syncDirectory %>' ],
-	compiledFiles: [ './+(tests|support)/**/*.d.ts', './+(tests|support)/**/*.js' ]
-};
-
-export const hexo = {
-	generate: {
-		src: '<%= siteDirectory %>',
-		dest: '<%= distDirectory %>'
-	}
-};
+	},
 
-export const intern = {
-	unit: {
-		options: {
-			runType: 'client',
-			config: '_build/tests/intern',
-			reporters: [
-				'Console', 'LcovHtml'
-			]
+	copy: {
+		'pretest-tutorials': {
+			src: 'site/source/tutorials/**/demo/**/*',
+			dest: '<%= buildDirectory %>/'
 		}
 	}
 };
 
-export const prompt = {
-	github: {
-		options: {
-			questions: [
-				{
-					config: 'github.username',
-					type: 'input',
-					message: 'Github username'
-				},
-				{
-					config: 'github.password',
-					type: 'password',
-					message: 'Github password'
-				}
-			]
-		}
-	}
+// Create config and task definitions for the tutorials to accomplish:
+//		-- run npm install for each tutorial demo
+//		-- run dojo test for each tutorial demo
+const tutorialsTestDir = buildDirectory + '/site/source/tutorials';
+export const tutorialLocations:TutorialDefinitions = {
+	tutorial001Initial: tutorialsTestDir + '/001_static_content/demo/initial/biz-e-corp',
+	tutorial001Finished: tutorialsTestDir + '/001_static_content/demo/finished/biz-e-corp',
+	tutorial002Initial: tutorialsTestDir + '/002_creating_an_application/demo/initial/biz-e-corp',
+	tutorial002Finished: tutorialsTestDir + '/002_creating_an_application/demo/finished/biz-e-corp',
+	tutorial003Initial: tutorialsTestDir + '/003_creating_widgets/demo/initial/biz-e-corp',
+	tutorial003Finished: tutorialsTestDir + '/003_creating_widgets/demo/finished/biz-e-corp',
+	tutorial004Initial: tutorialsTestDir + '/004_user_interactions/demo/initial/biz-e-corp',
+	tutorial004Finished: tutorialsTestDir + '/004_user_interactions/demo/finished/biz-e-corp',
+	tutorial005Initial: tutorialsTestDir + '/005_form_widgets/demo/initial/biz-e-corp',
+	tutorial005Finished: tutorialsTestDir + '/005_form_widgets/demo/finished/biz-e-corp',
+	tutorial006Initial: tutorialsTestDir + '/006_containers_and_injecting_state/demo/initial/biz-e-corp',
+	tutorial006DeployInitial: tutorialsTestDir + '/006_deploying_to_production/demo/initial/biz-e-corp'
 };
 
-export const publish = {
-	'gh-pages': {
-		options: {
-			branch: 'gh-pages',
-			cloneDirectory: '<%= distDirectory %>'
-		}
-	}
-};
+const npmConfigObj = gruntConfig[ 'npm-command' ] = <GruntConfig>{};
```

Comment:
I believe our preferred style is `{} as GruntConfig` vs `<GruntConfig> {}`, though I'm not sure that's captured in the styles guide in `dojo/meta` yet.